pkg/plugins/policies/meshhealthcheck/plugin/v1alpha1/plugin_test.go

```diff
@@ -188,14 +191,15 @@ healthChecks:
     path: /health
     requestHeadersToAdd:
       - header:
-          key: x-kuma-tags
+          key: X-Kuma-Tags
```

Comment:
I'm pretty sure this should be lower case. I think I would even add a validation to the base type to only accept lowercase headers.

Envoy lowercase headers by default (because of HTTP/2) and there were many problems with that. I would stick to lower case to avoid this altogether.

Reply:
I'd love to get @michaelbeaumont 's opinion here because I've been adding `http.CanonicalHeaderKey()` following his example.

Reply:
Ah I took this from the gateway plugin code.

Reply:
Going for lower case only allowed. Main reason is that it's easier to extend later than to restrict